lib/galaxy/model/store/__init__.py

```diff
@@ -2128,12 +2128,12 @@ def export_history(
         datasets = query.all()
         for dataset in datasets:
             dataset.annotation = get_item_annotation_str(sa_session, history.user, dataset)
-            add_dataset = (dataset.visible or include_hidden) and (not dataset.deleted or include_deleted)
+            should_include_file = (dataset.visible or include_hidden) and (not dataset.deleted or include_deleted)
             if dataset.id in self.collection_datasets:
-                add_dataset = True
+                should_include_file = True
 
             if dataset not in self.included_datasets:
-                self.add_dataset(dataset, include_files=add_dataset)
+                self.add_dataset(dataset, include_files=should_include_file)
```

Comment:
I think I would verify that the dataset exists at this spot here, but not as part of `add_dataset`.